test/controllers/TaskListControllerSpec.scala

```diff
@@ -106,6 +106,34 @@ class TaskListControllerSpec extends SpecBase with AppWithDefaultMockFixtures wi
       redirectLocation(result).value mustEqual controllers.routes.SessionExpiredController.onPageLoad().url
     }
 
+    "must redirect to declaration submit for a POST if declaration guaranteeAmendment" in {
+      setExistingUserAnswers(emptyUserAnswers.copy(status = SubmissionState.GuaranteeAmendment, departureId = Some(departureId)))
+
+      when(mockSubmissionConnector.postAmendment(any())(any())).thenReturn(response(OK))
+
+      val request = FakeRequest(POST, routes.TaskListController.onSubmit(lrn).url)
+
+      val result = route(app, request).value
+
+      status(result) mustEqual SEE_OTHER
+
+      redirectLocation(result).value mustEqual controllers.routes.DeclarationSubmittedController.onPageLoad(lrn).url
+    }
+
+    "must redirect to technical difficulties for a POST if declaration guaranteeAmendment response is not 2xx" in {
+      setExistingUserAnswers(emptyUserAnswers.copy(status = SubmissionState.GuaranteeAmendment, departureId = Some(departureId)))
+
+      when(mockSubmissionConnector.postAmendment(any())(any())).thenReturn(response(SEE_OTHER))
```

Comment:
Rather than returning a 303 from the connector we could create a 4xx/5xx generator. Something like
`val errorCode = Gen.choose(400, 599).sample.value`
or
```
forAll(Gen.choose(400, 599)) {
  errorCode =>
    ...
}
```